experiments/modern/src/App.js

```diff
@@ -20,6 +20,44 @@ async function getSkin() {
   return await initialize(zip, skinXml);
 }
 
+function GuiObjectEvents({ node, children }) {
+  return (
+    <div
+      onMouseDown={e => {
+        if (e.button === 2) {
+          node.js_trigger("onRightButtonDown", e.clientX, e.clientY);
+        } else {
+          node.js_trigger("onLeftButtonDown", e.clientX, e.clientY);
+        }
+      }}
+      onMouseUp={e => {
+        if (e.button === 2) {
+          node.js_trigger("onRightButtonUp", e.clientX, e.clientY);
+        } else {
+          node.js_trigger("onLeftButtonUp", e.clientX, e.clientY);
+        }
+      }}
+      onDoubleClick={e => {
+        if (e.button === 2) {
+          node.js_trigger("onRightButtonDblClk", e.clientX, e.clientY);
+        } else {
+          node.js_trigger("onLeftButtonDblClk", e.clientX, e.clientY);
+        }
+      }}
+      onMouseMove={e => node.js_trigger("onMouseMove", e.clientX, e.clientY)}
+      onMouseOver={e => node.js_trigger("onEnterArea", e.clientX, e.clientY)}
+      onMouseOut={e => node.js_trigger("onLeaveArea", e.clientX, e.clientY)}
```

Comment:
Can we use mouseenter/mouseleave? Those don't bubble, and thus we won't trigger extra events when going from one child to another.

Reply:
Oh yea, forgot about the differences in those calls, sounds good!